nxc/protocols/ssh.py

```diff
@@ -184,99 +186,104 @@ def check_if_admin_sudo(self):
-                self.logger.error(f"Internal Paramiko error for {username}:{process_secret(password)}, {e}")
+                key_content = self.read_key_file(private_key or self.args.key_file)
+                self.connect_with_key(username, password, key_content)
+                cred_type = "key"
             else:
-                self.logger.exception(e)
+                self.connect_with_password(username, password)
+                cred_type = "plaintext"
+            
+            cred_id = self.db.add_credential(cred_type, username, password, key=key_content)
+            shell_access = self.test_shell_access()
+            host_id = self.db.get_hosts(self.host)[0].id
+            if self.admin_privs:
+                self.logger.debug(f"User {username} logged in successfully and is root!")
+                self.db.add_admin_user(cred_type, username, password, host_id=host_id, cred_id=cred_id)
+            display_shell_access = f"({self.user_principal}) {self.server_os_platform} - {'Shell access!' if shell_access else 'Undetermined shell access.'}"
+            if self.args.key_file:
+                password = f"{process_secret(password)} (keyfile: {self.args.key_file})"
+            self.logger.success(f"{username}:{password} {self.mark_pwned()} {highlight(display_shell_access)}")
+            self.db.add_loggedin_relation(cred_id, host_id, shell=shell_access)
+            return shell_access
         except Exception as e:
-            self.logger.exception(e)
-            self.conn.close()
+            self.handle_connection_failure(e)
             return False
-        else:
-            shell_access = False
-            host_id = self.db.get_hosts(self.host)[0].id
+        finally:
+            self.conn.close()
 
-            if not stdout:
-                _, stdout, _ = self.conn.exec_command("whoami /priv")
-                stdout = stdout.read().decode(self.args.codec, errors="ignore")
-                self.server_os_platform = "Windows"
-                if "SeDebugPrivilege" in stdout:
-                    self.admin_privs = True
-                elif "SeUndockPrivilege" in stdout:
-                    self.admin_privs = True
-                    self.uac = "with UAC - "
+    def read_key_file(self, private_key):
+        if self.args.key_file:
+            with open(self.args.key_file) as f:
+                private_key = f.read()
+        return private_key
 
-            if not stdout:
-                self.logger.debug(f"User: {self.username} can't get a basic shell")
-                self.server_os_platform = "Network Devices"
-                shell_access = False
-            else:
-                shell_access = True
+    def connect_with_key(self, username, password, key_content):
+        self.logger.debug(f"Logging {self.host} with username: {self.username}, key: {self.args.key_file}")
+        pkey = paramiko.RSAKey.from_private_key(StringIO(key_content))
```

Comment:
What is with all the other key types? This should not be limited to RSA

Reply:
yeah good catch this is a last PR modification mistake, I'll fix that